### PR TITLE
use a textnode when rendering simple values

### PR DESCRIPTION
**Pull request by IagoLast**
When `value` is a simple value (number or boolean) is straightforward to create a text node.

Before this PR, `injectHTML` was being called for that case. This function creates a fragment and updates its `innerHTML` property to finally extract the child nodes, but when value is a number or a boolean it cannot be more than one single text node, so I don't see any reason for calling `injectHTML`.  (I think this is valid also when value is a string but it does not contain `<` or `>`)

Please correct me if I'm wrong, I'm just a noob exploring the code.

### CODE REVIEW

**Comment by WebReflection** [comment]
The virtual content is complicated and that `typeof` check aim is to actually opt out from most common cases that are [arrays](https://github.com/WebReflection/hyperHTML/blob/master/hyperhtml.js#L212) or [fragments](https://github.com/WebReflection/hyperHTML/blob/master/hyperhtml.js#L229) or regular [node](https://github.com/WebReflection/hyperHTML/blob/master/hyperhtml.js#L231) all with the intent to refresh the [childNodes](https://github.com/WebReflection/hyperHTML/blob/master/hyperhtml.js#L198)

Being the virtual content the worst solution out of 3 (text/content/virtualcontent) I discourage the usage of virtual content for numbers or booleans.

If you want those types to be part of the layout use at least one extra char around your value.

```js
// WRONG
render`User age is <strong>${user.age}</strong>`

// OK
render`User age is <strong> ${user.age} </strong>`
```

TL;DR your PR might work but it adds extra bytes to the mix without providing any real-word scenario, actually optimizing what would be a mistake since you don't want to mix up booleans, numbers, and strings that would be injected as HTML in there.

Whatever keeps consistent the intent and definition of the template will stay as such.

Sorry if this wasn't the answer you were looking for, I'm trying to tell people there are 3 things you can define in hyperHTML and 3 only, they wiull eventually get it.

'till that, no optimization for wrong patterns should be released. Thanks for your understanding.

**Comment by IagoLast** [comment]
Oh, so this is a correct optimization for a undesirable case right?

why not forbid to use certain types as virtual content?

something like:

```javascript
// ...
case 'boolean':
case 'number':
  console.error('If you want those types to be part of the layout use at least one extra char around your value...');
  break;
// ...
```

**Comment by WebReflection** [comment]
because it's forgiving but it cares about size. A huge console error that breaks the layout, when the problem is that they simply should understand how to properly write their layouts, is not an ideal solution.

Developers will need to understand how `hyperHTML` works 'cause it's something new and different.

Once I see people using it properly I'll think about micro optimizations.

Until then, I have way better things to do/think about 😉

**Comment by WebReflection** [comment]
P.S. also I wont accept PRs without tests, in case you'd like to contribute in the future, your PR was one with no tests, and 2 extra untested branches (I'm willing to have 100% coverage for branches too in here as well as in viperHTML, any help increasing coverage is welcome)
